bundles/GraphicServiceOrder/Controller/GraphicServiceOrderController.php

```diff
@@ -10,242 +10,285 @@
 
 namespace GraphicServiceOrder\Controller;
 
-use Gedmo\Sluggable\Util\Urlizer;
 use App\Service\JiraService;
 use Doctrine\ORM\EntityManagerInterface;
 use GraphicServiceOrder\Entity\GsOrder;
 use App\Service\OwnCloudService;
+use GraphicServiceOrder\Message\OwnCloudShare;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\Request;
 use GraphicServiceOrder\Form\GraphicServiceOrderForm;
 use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\HttpKernel\KernelInterface;
+use GraphicServiceOrder\Repository\GsOrderRepository;
 
 /**
  * Class GraphicServiceOrderController.
  */
 class GraphicServiceOrderController extends AbstractController
 {
-    private $jiraService;
-    private $ownCloudService;
-    private $formData;
-
-    /**
-     * GraphicServiceOrderController constructor.
-     *
-     * @param \App\Service\JiraService     $jiraService
-     * @param \App\Service\OwnCloudService $ownCloudService
-     *
-     * @TODO: Create service that extends AbstractJiraService instead of using JiraService.
-     */
-    public function __construct(
-        JiraService $jiraService,
-        OwnCloudService $ownCloudService
-    ) {
-        $this->jiraService = $jiraService;
-        $this->ownCloudService = $ownCloudService;
-    }
+  private $jiraService;
+  private $ownCloudService;
+  private $formData;
+  private $gsOrderRepository;
+  private $appKernel;
+
+  public function __construct(JiraService $jiraService, OwnCloudService $ownCloudService, GsOrderRepository $gsOrderRepository, KernelInterface $appKernel)
+  {
+    $this->jiraService = $jiraService;
+    $this->ownCloudService = $ownCloudService;
+    $this->gsOrderRepository = $gsOrderRepository;
+    $this->appKernel = $appKernel;
+  }
+
+  /**
+   * Create a service order.
+   *
+   * @Route("/new", name="graphic_service_order_form")
+   */
+  public function createOrder(EntityManagerInterface $entitityManagerInterface, Request $request, MessageBusInterface $bus)
+  {
+    $gsOrder = new GsOrder();
+    $form = $this->createForm(GraphicServiceOrderForm::class, $gsOrder);
+
+    $this->formData = [
+      'form' => $form->getData(),
+      'accounts' => $this->jiraService->getAllAccounts(),
+      'user' => $this->jiraService->getCurrentUser(),
+    ];
+
+    $form->handleRequest($request);
+
+    if ($form->isSubmitted() && $form->isValid()) {
+      // Do stuff on submission.
+
+      // Create a task on a jira project.
```

Comment:
All this create code should be moved to a service.
The controllers should be as slim as possible.